src/components/TextareaField/TextareaField.stories.tsx

```diff
@@ -0,0 +1,83 @@
+import { BADGE } from '@geometricpanda/storybook-addon-badges';
+import type { StoryObj, Meta } from '@storybook/react';
+import React from 'react';
+
+import { TextareaField } from './TextareaField';
+
+export default {
+  title: 'Components/TextareaField',
+  component: TextareaField,
+  args: {
+    placeholder: 'Enter long-form text here',
+    defaultValue: `Lorem ipsum, dolor sit amet consectetur adipisicing elit. Id neque nemo
+    dicta rerum commodi et fugiat quo optio veniam! Ea odio corporis nemo
+    praesentium, commodi eligendi asperiores quis dolorum porro.`,
+    label: 'Textarea Field',
+    rows: 5,
+    fieldNote: 'Longer Field description',
+  },
+  parameters: {
+    badges: [BADGE.BETA],
+  },
+} as Meta<Args>;
+
+type Args = React.ComponentProps<typeof TextareaField>;
+
+export const Default: StoryObj<Args> = {
+  render: (args) => (
+    <TextareaField aria-label="Text Label" {...args}></TextareaField>
+  ),
+};
+
+export const UsingChildren: StoryObj<Args> = {
+  args: {
+    children: `Lorem ipsum, dolor sit amet consectetur adipisicing elit. Id neque nemo
+    dicta rerum commodi et fugiat quo optio veniam! Ea odio corporis nemo
+    praesentium, commodi eligendi asperiores quis dolorum porro.`,
+    defaultValue: '',
+  },
+};
+
+export const WithNoDefaultValue: StoryObj<Args> = {
+  args: {
+    defaultValue: '',
+  },
+};
+
+export const WhenDisabled: StoryObj<Args> = {
+  args: {
+    disabled: true,
+    rows: 2,
+  },
+  parameters: {
+    axe: {
+      // Disabled input does not need to meet color contrast
+      disabledRules: ['color-contrast'],
+    },
+  },
+};
+
+export const WhenError: StoryObj<Args> = {
+  args: {
+    isError: true,
+    fieldNote: 'Text should be at least 100 characters',
+  },
+};
+
+export const WhenInvalid: StoryObj<Args> = {
+  args: {
+    maxLength: 10,
+  },
+};
+
+export const WhenRequired: StoryObj<Args> = {
+  args: {
+    required: true,
+  },
+};
+
+export const WithADifferentSize: StoryObj<Args> = {
+  args: {
+    rows: 10,
+  },
+};
```

Comment:
What do you think about having an interactive story to reflect the ongoing character count, like:
![image](https://user-images.githubusercontent.com/86632227/220750966-bdb48807-401d-4a28-b0f2-7c9320afcd60.png)

Reply:
not a bad idea. I was planning to create a separate ticket for the text count thing, as there are a few behaviors i want to discuss.

Reply:
@jinlee93 created ticket and tagged you on that one!